### PR TITLE
[pbw-4318] fix fullscreen ESC bug, added fallback if fullscreen API…

### PR DESCRIPTION
**Pull request by mistermendez**
…not supported natively

### CODE REVIEW

File: js/controller.js [hunk 1]
```diff
@@ -604,13 +612,17 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
       }
     },
 
-    onFullscreenChanged: function(event, fullscreen, paused) {
+    onFullscreenChanged: function() {
+      if (this.state.isFullScreenSupported) {
```
Comment (aldigun) [inline js/controller.js]: This IF block is checked again inside toggeFullScreen method: if (this.state.isFullScreenSupported)
We can refactor this.state.fullscreen = Fullscreen.isFullscreen; into toggleFullScreen()

File: js/controller.js [hunk 2]
```diff
@@ -687,15 +698,55 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
...
+
+      // unhide scroll bars
+      document.documentElement.style.overflow = 'visible';
+
+      //remove full window style
+      this.state.mainVideoWrapper.removeClass('fullscreen');
+    },
+
+    // exit full window on ESC key
+    exitFullWindowOnEscKey: function(event) {
```
Comment (aldigun) [inline js/controller.js]: This exit with ESC button is only applied to fullWindow. Does the other true fullScreen already has ESC button functionality?

Comment (mistermendez) [inline js/controller.js]: yup, native fullscreen already works with ESC

File: js/controller.js (hunk 1 above)
Comment (mistermendez) [inline js/controller.js]: that should work, let me update and test

Comment (mistermendez) [inline js/controller.js]: I'm getting this error - 

Failed to execute 'requestFullScreen' on 'Element': API can only be initiated by a user gesture.

**Comment by mistermendez** [comment]
@aldigun conflicts merged

**Comment by aldigun** [comment]
sweet!

**Pull request merged**